sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -155,6 +159,67 @@ dif_rstmgr_result_t dif_rstmgr_reset_info_clear(const dif_rstmgr_t *handle) {
   return kDifRstmgrOk;
 }
 
+dif_rstmgr_result_t dif_rstmgr_alert_info_set_enabled(
+    const dif_rstmgr_t *handle, dif_rstmgr_toggle_t state) {
+  if (handle == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  uint32_t enabled = (state == kDifRstmgrToggleEnabled) ? 0x1 : 0x0;
+
+  // This will clobber the `ALERT_INFO_CTRL.INDEX` field. However, the index
+  // field is only relevant during the crash dump read operation, and is
+  // set by the caller and not the hardware, so it is safe to clobber it.
+  mmio_region_write32(base_addr, RSTMGR_ALERT_INFO_CTRL_REG_OFFSET, enabled);
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_get_size(
+    const dif_rstmgr_t *handle, size_t *count) {
+  if (handle == NULL || count == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  *count = rstmgr_alert_info_dump_get_size(base_addr);
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_read(
+    const dif_rstmgr_t *handle, size_t index,
+    dif_rstmgr_alert_info_dump_value_t *value) {
+  if (handle == NULL || value == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  if (index >= rstmgr_alert_info_dump_get_size(base_addr)) {
+    return kDifRstmgrBadArg;
+  }
```

Comment:
I'm not sure this is a good API. Note that `BadArg` is incorrect here, since this performs an MMIO operation.

I would go with expecting the caller to not pass an out-of-bounds length. Alternatively, you could move to a transactional iterator API where a caller begins a crash dump read and repeatedly calls `end()` to get the next word in the dump.

Reply:
> Note that BadArg is incorrect here

Well, there is an MMIO read but AFAICT it has no side-effects because the register is read-only.

Reply:
Sorry, we've tried to be clear here, the read itself is a side-effect.

For simplicitly, this should be `kDifRstmgrError` if you don't want to add another variant/enum.